packages/react/src/date-range-picker/DateRangePicker.tsx

```diff
@@ -174,96 +178,74 @@ export const DateRangePicker = ({
 				return;
 			}
 		},
-		[closeCalendar, inputMode, onChange, valueAsDateOrUndefined]
+		[closeCalendar, inputMode, onChange, valueAsDateOrUndefined, dateFormat]
 	);
 
 	// From input state
 	const [fromInputValue, setFromInputValue] = useState(
-		transformValuePropToInputValue(value.from)
+		transformValuePropToInputValue(value.from, dateFormat)
 	);
 
-	const debouncedAttemptDateParseFromDate = useDebouncedCallback(
-		(inputValue: string) => {
-			// Ensure the text entered is a valid date
-			const parsedDate = parseDate(inputValue);
-			const constrainedDate = constrainDate(parsedDate, minDate, maxDate);
-
-			const nextValue = ensureValidDateRange({
-				from: constrainedDate,
-				to: valueAsDateOrUndefined.to,
-			});
-
-			// When there is no value OR there is a valid date, only trigger the `onChange` callback
-			// `onInputChange` will not be called
-			if (!inputValue || constrainedDate) {
-				onChange(nextValue);
-				return;
-			}
+	const onFromInputBlur = (e: FocusEvent<HTMLInputElement>) => {
+		const inputValue = e.target.value;
+
+		// Ensure the text entered is a valid date
+		const parsedDate = parseDate(inputValue);
+		const constrainedDate = constrainDate(parsedDate, minDate, maxDate);
 
-			// Trigger the callbacks
+		const nextValue = ensureValidDateRange({
+			from: constrainedDate,
+			to: valueAsDateOrUndefined.to,
+		});
+
+		if (!inputValue || constrainedDate) {
 			onChange(nextValue);
+		} else {
 			onFromInputChangeProp?.(inputValue);
```

Comment:
Are we handling the usecase when someone enters something that isn't at all a date like "cat"?

Reply:
Yep, if you look at the log in https://design-system.agriculture.gov.au/pr-preview/pr-1563/components/date-picker#tracking-the-input-value, you'll see that `cat` gets logged so consumers can run validation like we do in that example